web/src/engine/websites/DynastyScans.ts

```diff
@@ -0,0 +1,68 @@
+import { Tags } from '../Tags';
+import icon from './DynastyScans.webp';
+import { Chapter, DecoratableMangaScraper, Manga,type MangaPlugin, Page } from '../providers/MangaPlugin';
+import { FetchJSON, FetchRequest } from '../FetchProvider';
+import * as Common from './decorators/Common';
+
+@Common.MangaCSS(/^https?:\/\/dynasty-scans\.com\/[^/]+/, 'h2.tag-title b')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('dynasty-scans', `DynastyScans`, 'https://dynasty-scans.com' , Tags.Language.English, Tags.Media.Manga, Tags.Source.Scanlator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangalist = [];
+        const categories = ['/series', '/anthologies', '/issues', '/doujins'];
+        for (const cat of categories) {
+            const mangas = await this.getMangasFromCategory(provider, cat);
+            mangas.length > 0 ? mangalist.push(...mangas) : false;
+        }
+        return mangalist;
+    }
+
+    async getMangasFromCategory(provider: MangaPlugin, cat: string): Promise<Manga[]> {
+        const mangas = [];
+        for (let i = 1, run = true; run; i++) {
+            try {
+                const json = await FetchJSON(new FetchRequest(new URL(`${cat}.json?page=${i}`, this.URI).href));

```

Comment:
Make a type definition for the JSON data structure

Reply:
Like for Digital team, i thought about it but objects keys are not fixed at some point, dont know howi could create type in these conditions. Thats why i used ['this'] notation.

![image](https://user-images.githubusercontent.com/10671518/214251398-14e96c33-6f51-43e7-a10b-552ba18f99da.png)